core/src/main/java/io/undertow/server/BasicSSLSessionInfo.java

```diff
@@ -67,7 +67,15 @@ public BasicSSLSessionInfo(byte[] sessionId, String cypherSuite, String certific
             int i=0;
             for(java.security.cert.Certificate cert : certCol) {
                 this.peerCertificate[i] = cert;
-                this.certificate[i++] = X509Certificate.getInstance(cert.getEncoded());
+                try {
+                    this.certificate[i] = X509Certificate.getInstance(cert.getEncoded());
+                } catch (CertificateException ce) {
+                    // [UNDERTOW-1969] We don't care about deprecated JDK methods failure caused by the fact newer JDKs
+                    // doesn't support them anymore. "this.certificate" is used only by deprecated method
+                    // {@link SSLSessionInfo.getPeerCertificateChain()} which call should be avoided by API users.
+                    this.certificate[i] = null;
```

Comment:
@ropalka @fl4via This is not a big deal but if this happens shouldn't the result be that this.certificate (the whole array) gets set to null? Then getPeerCertificateChain() will throw SSLPeerUnverifiedException which seems cleaner than returning an array with some null elements.

Reply:
In case you agree I sent up https://github.com/undertow-io/undertow/pull/1255

Reply:
I agree.

Reply:
@bstansberry  thanks for fixing this!